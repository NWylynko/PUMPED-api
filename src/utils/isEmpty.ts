export const isArrayEmpty = (array: any[]): true | undefined => (array.length === 0 ? true : undefined);

export const isEmpty = (object: any): true | undefined => (!object ? true : undefined);
