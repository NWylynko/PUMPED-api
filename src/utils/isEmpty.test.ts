import { isArrayEmpty, isEmpty } from './isEmpty';

describe('array', () => {
  it('is empty', () => {
    const array: any[] = [];

    expect(isArrayEmpty(array)).toMatchSnapshot();
  });
  it('it is not empty', () => {
    const array: any[] = ['test'];

    expect(isArrayEmpty(array)).toMatchSnapshot();
  });
});

describe('object', () => {
  it('is empty', () => {
    const object: any = null;

    expect(isEmpty(object)).toMatchSnapshot();
  });
  it('it is not empty', () => {
    const object: any = {
      not: 'empty',
    };

    expect(isEmpty(object)).toMatchSnapshot();
  });
});
