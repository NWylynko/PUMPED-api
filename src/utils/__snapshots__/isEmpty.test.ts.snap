// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array is empty 1`] = `true`;

exports[`array it is not empty 1`] = `undefined`;

exports[`object is empty 1`] = `true`;

exports[`object it is not empty 1`] = `undefined`;
